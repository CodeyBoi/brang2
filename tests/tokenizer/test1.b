print "input: ";

getchar a;
getchar b;

if a < b {
    print "a<b\n";
} else if a > b {
    print "a>b\n";
} else {
    print "a=b\n";
}
