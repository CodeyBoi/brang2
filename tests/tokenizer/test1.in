print "input: ";

let a = 2;
let b = 5;

if a < b {
    print "a<b\n";
} else if a > b {
    print "a>b\n";
} else {
    print "a=b\n";
}
