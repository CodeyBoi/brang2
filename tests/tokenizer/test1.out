Print
String("input: ")
Semicolon
Let
Identifier("a")
Equal
Number(2)
Semicolon
Let
Identifier("b")
Equal
Number(5)
Semicolon
If
Identifier("a")
Less
Identifier("b")
LeftBrace
Print
String("a<b\n")
Semicolon
RightBrace
Else
If
Identifier("a")
Greater
Identifier("b")
LeftBrace
Print
String("a>b\n")
Semicolon
RightBrace
Else
LeftBrace
Print
String("a=b\n")
Semicolon
RightBrace
